<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>turnarond</author>
 <class>DeployMaster</class>
 <widget class="QMainWindow" name="DeployMaster">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>916</width>
    <height>832</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DeployMaster - FTP &amp; Telnet 批量运维工具</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: #F0F0F0; font-family: &quot;Microsoft YaHei&quot;, &quot;Segoe UI&quot;;</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="0,3,0,1">
    <property name="leftMargin">
     <number>12</number>
    </property>
    <property name="topMargin">
     <number>10</number>
    </property>
    <property name="rightMargin">
     <number>12</number>
    </property>
    <property name="bottomMargin">
     <number>12</number>
    </property>
    <item>
     <widget class="QFrame" name="frame_commonConfig">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_common" stretch="2,1">
       <property name="spacing">
        <number>10</number>
       </property>
       <property name="leftMargin">
        <number>8</number>
       </property>
       <property name="topMargin">
        <number>6</number>
       </property>
       <property name="rightMargin">
        <number>8</number>
       </property>
       <property name="bottomMargin">
        <number>6</number>
       </property>
       <item>
        <widget class="QGroupBox" name="groupBox_ipList">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>100</height>
          </size>
         </property>
         <property name="title">
          <string>目标IP列表（每行一个）</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_ipList">
          <property name="spacing">
           <number>4</number>
          </property>
          <property name="leftMargin">
           <number>6</number>
          </property>
          <property name="topMargin">
           <number>4</number>
          </property>
          <property name="rightMargin">
           <number>6</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QTextEdit" name="txt_ipList">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>70</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>70</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Microsoft YaHei</family>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="lineWrapMode">
             <enum>QTextEdit::LineWrapMode::NoWrap</enum>
            </property>
            <property name="plainText">
             <string>192.168.2.124
192.168.2.126
192.168.2.127
192.168.2.128</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_auth">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>100</height>
          </size>
         </property>
         <property name="title">
          <string>认证与路径</string>
         </property>
         <layout class="QFormLayout" name="formLayout_auth">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::FieldGrowthPolicy::AllNonFixedFieldsGrow</enum>
          </property>
          <property name="rowWrapPolicy">
           <enum>QFormLayout::RowWrapPolicy::DontWrapRows</enum>
          </property>
          <property name="verticalSpacing">
           <number>6</number>
          </property>
          <property name="leftMargin">
           <number>6</number>
          </property>
          <property name="topMargin">
           <number>4</number>
          </property>
          <property name="rightMargin">
           <number>6</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label_user">
            <property name="text">
             <string>用户名：</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="txt_user">
            <property name="text">
             <string>root</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_pass">
            <property name="text">
             <string>密码：</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="txt_pass">
            <property name="text">
             <string>SE-2025</string>
            </property>
            <property name="echoMode">
             <enum>QLineEdit::EchoMode::Password</enum>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_defaultRemotePath">
            <property name="text">
             <string>默认远程路径：</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="txt_defaultRemotePath">
            <property name="text">
             <string>/apps/m580cn/bin/</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabPosition">
       <enum>QTabWidget::TabPosition::North</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_deploy">
       <attribute name="title">
        <string>批量部署</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_deploy" columnstretch="1,3,2,0">
        <item row="0" column="0" colspan="3">
         <widget class="QGroupBox" name="groupBox_upload">
          <property name="acceptDrops">
           <bool>true</bool>
          </property>
          <property name="title">
           <string>上传内容（支持拖拽）</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_upload">
           <item>
            <widget class="QLabel" name="label_dragHint">
             <property name="styleSheet">
              <string notr="true">color: gray; font-style: italic;</string>
             </property>
             <property name="text">
              <string>点击“添加”或拖拽文件/文件夹到这里</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="DropListWidget" name="list_uploadedItems">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>100</height>
              </size>
             </property>
             <property name="acceptDrops">
              <bool>true</bool>
             </property>
             <property name="dragDropMode">
              <enum>QAbstractItemView::DragDropMode::DropOnly</enum>
             </property>
             <property name="selectionMode">
              <enum>QAbstractItemView::SelectionMode::ExtendedSelection</enum>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_uploadBtns">
             <item>
              <spacer name="horizontalSpacer_upload">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="btn_addFiles">
               <property name="minimumSize">
                <size>
                 <width>90</width>
                 <height>22</height>
                </size>
               </property>
               <property name="text">
                <string>添加文件...</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_addFolders">
               <property name="minimumSize">
                <size>
                 <width>90</width>
                 <height>22</height>
                </size>
               </property>
               <property name="text">
                <string>添加文件夹...</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_clearUploadList">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>22</height>
                </size>
               </property>
               <property name="text">
                <string>清空</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0" colspan="3">
         <widget class="QGroupBox" name="groupBox_ftp">
          <property name="title">
           <string>FTP 设置</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_ftp">
           <item row="0" column="1" colspan="2">
            <widget class="QLineEdit" name="txt_remotePath">
             <property name="text">
              <string>/apps/m580cn/bin/</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QCheckBox" name="chk_rebootAfterDeploy">
             <property name="text">
              <string>部署后重启设备</string>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_remotePath">
             <property name="text">
              <string>远程路径：</string>
             </property>
            </widget>
           </item>
           <item row="0" column="4">
            <widget class="QCheckBox" name="chk_clearRemoteBeforeUpload">
             <property name="text">
              <string>部署前清空远程路径</string>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <layout class="QHBoxLayout" name="horizontalLayout_deployBtns">
          <item>
           <widget class="QPushButton" name="btn_deploy">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>30</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: #4CAF50; color: white; font-weight: bold;</string>
            </property>
            <property name="text">
             <string>开始部署</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_clearLog">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>30</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: #9E9E9E; color: white;</string>
            </property>
            <property name="text">
             <string>清除日志</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_deploy">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item row="0" column="3" rowspan="4">
         <widget class="QGroupBox" name="groupBox_remotePreview">
          <property name="title">
           <string>远端预览</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_remote">
           <item>
            <widget class="QComboBox" name="cmb_targetIPs">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>180</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_refreshRemote">
             <property name="maximumSize">
              <size>
               <width>60</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>刷新</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox_remoteFiles">
             <property name="title">
              <string>远程文件</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_tree">
              <item>
               <widget class="QTreeView" name="tree_remoteFiles">
                <property name="headerHidden">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame_splitter">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>5</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::Shape::HLine</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Sunken</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_log">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>220</height>
       </size>
      </property>
      <property name="title">
       <string>综合日志输出</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_log">
       <item>
        <widget class="QTextEdit" name="txt_globalLog">
         <property name="font">
          <font>
           <family>Microsoft YaHei</family>
           <pointsize>11</pointsize>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: black; color: #00FF00;</string>
         </property>
         <property name="lineWrapMode">
          <enum>QTextEdit::LineWrapMode::WidgetWidth</enum>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="plainText">
          <string>[日志] 系统启动完成，等待操作...</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>916</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="action_new"/>
    <addaction name="action_open"/>
    <addaction name="action_save"/>
    <addaction name="separator"/>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_deploy">
    <property name="title">
     <string>部署</string>
    </property>
    <addaction name="action_start_deploy"/>
    <addaction name="action_stop_deploy"/>
    <addaction name="action_pause_deploy"/>
   </widget>
   <widget class="QMenu" name="menu_help">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="action_about"/>
   </widget>
   <addaction name="menu_file"/>
   <addaction name="menu_deploy"/>
   <addaction name="menu_help"/>
  </widget>
  <widget class="QToolBar" name="toolbar">
   <property name="windowTitle">
    <string>工具栏</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_start_deploy"/>
   <addaction name="action_stop_deploy"/>
   <addaction name="separator"/>
   <addaction name="action_clear_log"/>
  </widget>
  <action name="action_new">
   <property name="text">
    <string>新建</string>
   </property>
  </action>
  <action name="action_open">
   <property name="text">
    <string>打开</string>
   </property>
  </action>
  <action name="action_save">
   <property name="text">
    <string>保存</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="text">
    <string>退出</string>
   </property>
  </action>
  <action name="action_start_deploy">
   <property name="icon">
    <iconset theme="media-playback-start"/>
   </property>
   <property name="text">
    <string>开始部署</string>
   </property>
  </action>
  <action name="action_stop_deploy">
   <property name="icon">
    <iconset theme="media-playback-stop"/>
   </property>
   <property name="text">
    <string>停止部署</string>
   </property>
  </action>
  <action name="action_pause_deploy">
   <property name="icon">
    <iconset theme="media-playback-pause"/>
   </property>
   <property name="text">
    <string>暂停部署</string>
   </property>
  </action>
  <action name="action_about">
   <property name="text">
    <string>关于</string>
   </property>
  </action>
  <action name="action_clear_log">
   <property name="text">
    <string>清除日志</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DropListWidget</class>
   <extends>QListWidget</extends>
   <header>droplistwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
